proposals/0035-linalg-matrix.md

```diff
@@ -641,6 +687,18 @@ expected target data format. When read from `groupshared` memory, the data may
 be in any arithmetic or packed data type. If the type mismatches the target data
 type of the matrix a data conversion is applied on load.
 
+The following table specifies the valid values for the `Layout` parameter
+given the `Load` method type and matrix scope.  All other combinations are
+unsupported:
+
+| Operation | Matrix Scope | Matrix Layout |
+|-----------|--------------|------------|
+| `Matrix::Load(ByteAddressBuffer)` | `Thread` | any |
+| `Matrix::Load(*)` | `Wave` | `RowMajor`, `ColMajor` |
+
+If the `Layout` argument is `MulOptimal` or `OuterProductOptimal`, then
+the `Stride` must be `0`.

```

Comment:
This would be nice to enforce at the language level, ideally encoded into the function signature but if that isn't possible then in can be deferred and handled in a validation step.

Either way it should be noted that the implementation should enforce this, we shouldn't rely on users just knowing to only put 0